frontend/src/pages/RecipientRecord/pages/CommunicationLog.js

```diff
@@ -1,8 +1,97 @@
-import React from 'react';
+import React, { useState, useContext } from 'react';
 import PropTypes from 'prop-types';
 import { Helmet } from 'react-helmet';
 
-export default function CommunicationLog({ recipientName, regionId }) {
+import { Link } from 'react-router-dom';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import { getCommunicationLogsByRecipientId } from '../../../fetchers/communicationLog';
+import AppLoadingContext from '../../../AppLoadingContext';
+import WidgetContainer from '../../../components/WidgetContainer';
+import CommunicationLogTable from './components/CommunicationLogTable';
+
+const COMMUNICATION_LOG_PER_PAGE = 10;
+
+export default function CommunicationLog({ recipientName, regionId, recipientId }) {
+  const [logs, setLogs] = useState();
+  const [error, setError] = useState();
+  const [sortConfig, setSortConfig] = useState({
+    sortBy: 'communicationDate',
+    direction: 'desc',
+    offset: 0,
+    activePage: 1,
+  });
+
+  const { setIsAppLoading } = useContext(AppLoadingContext);
+
+  useDeepCompareEffect(() => {
+    async function fetchLogs() {
+      try {
+        setError(null);
+        setIsAppLoading(true);
+        const response = await getCommunicationLogsByRecipientId(
+          String(regionId),
+          String(recipientId),
+          sortConfig.sortBy,
+          sortConfig.direction,
+          sortConfig.offset,
+        );
+
+        setLogs(response);
+      } catch (err) {
+        setError('Error fetching communication logs');
+      } finally {
+        setIsAppLoading(false);
+      }
+    }
+    fetchLogs();
+  }, [
+    logs,
```

Comment:
I think this can be removed. Since logs are updated by the fetch, `fetchLogs` happens twice on page load

Reply:
Err, I did it again. commented about something kinda unrelated to this PR, sorry! I'll resolve this one

Reply:
Easy fix, I will include it here